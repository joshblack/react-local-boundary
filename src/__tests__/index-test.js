/* @noflow */

import React from 'react';
import PropTypes from 'prop-types';
import {mount} from 'enzyme';

describe('LocalBoundary Component', () => {
  let LocalBoundary;

  beforeEach(() => {
    LocalBoundary = require('../').default;
  });

  it('should render a valid child component', () => {
    const tree = mount(
      <LocalBoundary renderError={jest.fn()}>
        <div />
      </LocalBoundary>,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should render the error handler for an invalid subtree', () => {
    const error = new Error('Error');
    const mockErrorComponent = jest.fn(error => <div />);
    const ErrorComponent = () => {
      throw error;
    };

    mount(
      <LocalBoundary renderError={mockErrorComponent}>
        <ErrorComponent />
      </LocalBoundary>,
    );

    expect(mockErrorComponent).toHaveBeenCalledWith(error);
  });

  it('should preserve any context defined in the current subtree', () => {
    class ContextProvider extends React.Component {
      static propTypes = {
        children: PropTypes.node,
      };
      static childContextTypes = {
        foo: PropTypes.string,
      };
      getChildContext = () => ({foo: 'bar'});
      render() {
        return this.props.children;
      }
    }

    const contextValues = [];
    const ContextChild = (_, context) => {
      contextValues.push(context);
      return <div />;
    };
    ContextChild.contextTypes = {
      foo: PropTypes.string,
    };

    mount(
      <ContextProvider>
        <LocalBoundary renderError={jest.fn()}>
          <ContextChild />
        </LocalBoundary>
      </ContextProvider>,
    );

    expect(contextValues).toEqual([{foo: 'bar'}]);
  });
});
