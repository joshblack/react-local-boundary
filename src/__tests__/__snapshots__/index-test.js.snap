// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalBoundary Component should render a valid child component 1`] = `
<LocalBoundary
  renderError={[Function]}
>
  <div />
</LocalBoundary>
`;
